owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsSecurityFragment.kt

```diff
@@ -0,0 +1,218 @@
+
+                // Do not allow to use biometric lock since Pattern lock has been disabled
+                disableBiometric(getString(R.string.prefs_biometric_summary))
+            } else {
+                showMessageInSnackbar(getString(R.string.pattern_error_remove))
+            }
+        }
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        setPreferencesFromResource(R.xml.settings_security, rootKey)
+
+        screenSecurity = findPreference(SCREEN_SECURITY)
+        prefPasscode = findPreference(PassCodeActivity.PREFERENCE_SET_PASSCODE)
+        prefPattern = findPreference(PatternLockActivity.PREFERENCE_SET_PATTERN)
+        prefBiometric = findPreference(BiometricActivity.PREFERENCE_SET_BIOMETRIC)
+        prefTouchesWithOtherVisibleWindows = findPreference(PREFERENCE_TOUCHES_WITH_OTHER_VISIBLE_WINDOWS)
+
+        // Passcode lock
+        prefPasscode?.setOnPreferenceChangeListener { preference: Preference?, newValue: Any ->
+            if (settingsViewModel.isPatternSet()) {
+                showMessageInSnackbar(getString(R.string.pattern_already_set))
+            } else {
+                val intent = Intent(activity, PassCodeActivity::class.java)
+                if (newValue as Boolean) {
+                    intent.action = PassCodeActivity.ACTION_REQUEST_WITH_RESULT
+                    enablePasscodeLauncher.launch(intent)
+                } else {
+                    intent.action = PassCodeActivity.ACTION_CHECK_WITH_RESULT
+                    disablePasscodeLauncher.launch(intent)
+                }
+            }
+            false
+        }
+
+        // Pattern lock
+        prefPattern?.setOnPreferenceChangeListener { preference: Preference?, newValue: Any ->
+            if (settingsViewModel.isPasscodeSet()) {
+                showMessageInSnackbar(getString(R.string.passcode_already_set))
+            } else {
+                val intent = Intent(activity, PatternLockActivity::class.java)
+                if (newValue as Boolean) {
+                    intent.action = PatternLockActivity.ACTION_REQUEST_WITH_RESULT
+                    enablePatternLauncher.launch(intent)
+                } else {
+                    intent.action = PatternLockActivity.ACTION_CHECK_WITH_RESULT
+                    disablePatternLauncher.launch(intent)
+                }
+            }
+            false
+        }
+
+        // Biometric lock
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            screenSecurity?.removePreference(prefBiometric)
+        } else if (prefBiometric != null) {
+            biometricManager = BiometricManager.getBiometricManager(activity)
+
+            // Disable biometric lock if Passcode or Pattern locks are disabled
+            if (prefPasscode?.isChecked == false && prefPattern?.isChecked == false) {
+                disableBiometric(getString(R.string.prefs_biometric_summary))
+            }
```

Comment:
Is this relevant? Is there a scenario where the biometric could be enabled, and passcode and pattern not? Just wondering 🤔

Reply:
This is only executed when the fragment is inflated. It's not that biometric is enabled/checked, but we have to disable it as by default, the initial state of the `CheckBoxPreference` is enabled. If we had chosen to set it to disabled by default in the XML, we would have to check the opposite: if passcode or pattern are checked, then enable biometric. I think it's easier this way, but let me know if you think the other way is better

Reply:
Btw, I took advantage of this and revised the `disableBiometric` function. Now it doesn't receive any parameter, it was unnecessary 😁